OpenEdXMobile/src/main/java/org/edx/mobile/util/BrowserUtil.java

```diff
@@ -87,8 +90,15 @@ private static void openInBrowser(FragmentActivity context, String url) {
         intent.addCategory(Intent.CATEGORY_BROWSABLE);
         intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
         intent.setData(Uri.parse(url));
-        context.startActivity(intent);
-
+        try {
+            context.startActivity(intent);
+        } catch (ActivityNotFoundException e) {
+            Toast.makeText(context, R.string.cannot_open_url, Toast.LENGTH_SHORT).show();
+
+            // Send non-fatal exception
+            logger.error(new Exception(String.format("No activity found (browser cannot handle request) for this url: %s, error:\n", url)
+                    + e.getMessage()), true);
+        }
         AnalyticsRegistry analyticsRegistry = environment.getAnalyticsRegistry();
```

Comment:
is it necessary to send the analytics on raising `ActivityNotFoundException`?

Reply:
Nops, it should come after '`context.startActivity(intent);`' this line, let me correct it.